hasher-matcher-actioner/hmalib/fetching/bank_store.py

```diff
@@ -0,0 +1,289 @@
+        # full expected state of a bank-member.
+        signal_type_view = self._api_cls.naive_convert_to_signal_type(
+            self._signal_types, self.collab.to_pytx_collab_config(), {key: value}
+        )
+
+        if len(signal_type_view) == 0:
+            # Cases where none of the signal types in the update can be
+            # processed. eg. NCMEC returning pdna
+            return
+
+        # Ascertain that all signal_types map from the same content type.
+        # HMA at present can't store multiple content types in one bank
+        # member.
+        unique_content_types = set(
+            # Is there a signal type that may have multiple content_types?
+            list(map(lambda st: st.get_content_types()[0], signal_type_view.keys()))
+        )
+        if len(unique_content_types) > 1:
+            raise MultipleContentTypesInFetchDeltaRow()
+
+        content_type = unique_content_types.pop()
+
+        opinion_lists = [
+            metadata.get_as_opinions()
+            for val in signal_type_view.values()
+            for metadata in val.values()
+        ]
+        unique_opinion_tags = set()
+        for opinion_list in opinion_lists:
+            unique_opinion_tags.update(self.convert_opinions_to_label(opinion_list))
+
+        try:
+            member = self.banks_table.add_keyed_bank_member(
+                self._import_bank.bank_id,
+                str_key,
+                content_type,
+                None,
+                None,
+                None,
+                "",
+                True,
+                unique_opinion_tags,
+            )
+        except KeyError:
+            # member already exists, update labels just to be sure.
+            member = self.banks_table.update_bank_member(
+                self.banks_table._key_for_bank(self._import_bank.bank_id, str_key),
+                notes="",
+                bank_member_tags=unique_opinion_tags,
+            )
+
+        for signal_type in signal_type_view:
+            for hash_value in signal_type_view[signal_type].keys():
+                self.banks_table.add_bank_member_signal(
+                    bank_id=self._import_bank.bank_id,
+                    bank_member_id=member.bank_member_id,
+                    signal_type=signal_type,
+                    signal_value=hash_value,
+                )
+
+                # TODO: We don't verify that all existing bank-member signals should continue existing.
+
+    def _handle_delete(self, key: TUpdateRecordKey):
+        """
+        Respond to a delete from a signal exchange API. Typically, this would
+        happen at most once per key per flush.
+        """
+        str_key = self._coerce_key(key)
+        member = self.banks_table.get_keyed_bank_member(
+            self._import_bank.bank_id, str_key
+        )
+        if member:
+            self.banks_table.remove_bank_member(member.bank_member_id)
+
+    def convert_opinions_to_label(self, opinions: t.Sequence[SignalOpinion]):
```

Comment:
blocking: I'm uncomfortable with automatically taking in tags from APIs and putting them in directly as labels of the same category as the users. It's possible that the user takes a strong action in some case for matches that can be unexpectedly triggered when they join a collaboration. I think the user needs to retain control of the action label namespace the whole time, and we should either just discard the tags for now, or if you want to keep them, namespace them with a string that's really unlikely to collide with the users -

collaboration:<collab_name>:tag

Reply:
boop

Reply:
This is a quick follow. #1220